test/IntegrationTests/StartupHookTests.cs

```diff
@@ -131,5 +132,14 @@ private void AssertExpectationsMet(List<WebServerSpanExpectation> expectations,
         string finalMessage = Environment.NewLine + string.Join(Environment.NewLine, failures.Select(f => " - " + f));
         Assert.True(!failures.Any(), finalMessage);
     }
+
+    private IImmutableList<IMockSpan> RunTestApplication(bool enableStartupHook = true)
+    {
+        int agentPort = TcpPortProvider.GetOpenPort();
+        using var agent = new MockZipkinCollector(Output, agentPort);
```

Comment:
Not related to the PR but it seems that the call to `TcpPortProvider.GetOpenPort()` should actually be done inside the mock collector and the port parameter be a nullable. Most tests would just do like this one and use `agent.Port` as needed.